services/spar/src/Spar/Sem/GalleyAccess/Http.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Spar.Sem.GalleyAccess.Http where
+
+import Bilge
+import Control.Monad.Except
+import Imports hiding (log)
+import Polysemy
+import Polysemy.Error
+import Spar.Error (SparError)
+import Spar.Intra.Brig (MonadSparToBrig (..))
+import Spar.Intra.Galley (MonadSparToGalley)
+import qualified Spar.Intra.Galley as Intra
+import Spar.Sem.GalleyAccess
+import qualified System.Logger as Log
+import qualified System.Logger.Class as LogClass
+
+data RunHttpEnv = RunHttpEnv
+  { rheLogger :: Log.Logger,
+    rheManager :: Bilge.Manager,
+    rheRequest :: Bilge.Request
+  }
+
+newtype RunHttp a = RunHttp
+  { unRunHttp :: ReaderT RunHttpEnv (ExceptT SparError (HttpT IO)) a
+  }
+  deriving newtype (Functor, Applicative, Monad, MonadError SparError, MonadIO, MonadHttp, MonadReader RunHttpEnv)
+
+viaRunHttp ::
+  Members '[Error SparError, Embed IO] r =>
+  RunHttpEnv ->
+  RunHttp a ->
+  Sem r a
+viaRunHttp env m = do
+  ma <- embed @IO $ runHttpT (rheManager env) $ runExceptT $ flip runReaderT env $ unRunHttp m
+  case ma of
+    Left err -> throw err
+    Right a -> pure a
+
+instance LogClass.MonadLogger RunHttp where
+  log lvl msg = do
+    logger <- asks rheLogger
+    Log.log logger lvl msg
+
+instance MonadSparToGalley RunHttp where
+  call modreq = do
+    req <- asks rheRequest
+    httpLbs req modreq
+
+instance MonadSparToBrig RunHttp where
+  call modreq = do
+    req <- asks rheRequest
+    httpLbs req modreq
```

Comment:
This is a bit weird: the two instances only do what they should because they are carrying the correct resp. hostnames in the `RunHttpEnv` i the reader.  Nothing guarantees that's the case.  Right?  Time to kill these classes, I think...

(I give you the original code doesn't look much better.  Polysemy's an improvement!  :))

Reply:
Yeah, things are currently a bit janky here. I'm trying to change the implementation code as little as possible in order to get the polysemization done ASAP. Getting everything under the polysemy umbrella is the next thing on the critical path here; once it's out of the way, we can do cleanup and testing in parallel.